src/book/02-system/01-nubots/10-mathematics.mdx

```diff
@@ -0,0 +1,258 @@
+
+When drawing a vector $\mathbf{r}_{AB}^b$, that is a vector from $B$ to $A$ in the space $b$, the **tail** is at $B$ and the **head** is at $A$. **Vector addition** requires that the head of one vector is the same as the tail of the other. This is demonstrated [here](http://mathworld.wolfram.com/VectorAddition.html). We have that
+
+$$
+\mathbf{r}^b_{AC} = \mathbf{r}^b_{AB} + \mathbf{r}^b_{BC} = \mathbf{r}^b_{BC} + \mathbf{r}^b_{AB}
+$$
+
+since vector addition is **commutative**. This is written in code as
+
+```cpp
+Eigen::Vector3d rACb = rABb + rBCb;
+```
+
+## Rotations
+
+To describe the motion of the robot, we need a **reference frame** and a **coordinate system**.
+
+The reference frame is the **physical rigid body** and the coordinate system is a **mathematical concept**, which is represented as a **basis**.
+
+NUbots use a **right-handed coordinate system** to measure positions and rotations.
+
+![A right hand with the index finger pointed forwards and labelled 'x', the middle finger pointing to the left and labelled 'y', and the thumb pointing upwards and labelled 'z'](./images/right-hand-rule.svg 'Right-Hand Rule. Image (modified) from Wikimedia Commons https://en.wikipedia.org/wiki/File:Right_hand_rule_cross_product.svg')
+
+**Rotations** are defined by a 3x3 matrix. If we have a vector $\mathbf{r}^a$, that is the vector $\mathbf{r}$ in the coordinate system $a$, the rotation
+
+$$
+\begin{bmatrix}
+\cos(\psi) & -\sin(\psi) & 0 \\
+\sin(\psi) & \cos(\psi) & 0 \\
+0 & 0 & 1 \end{bmatrix}
+$$
+
+will rotate $\mathbf{r}^a$ by $\psi$ radians around the coordinate axis $\mathbf{a}_3$.
+
+More on rotation matrices can be found [here](http://mathworld.wolfram.com/RotationMatrix.html).
+
+The orientation of the basis $b$ relative to $a$ can be described by **three consecutive rotations** about the main axis. There are 12 different orders to do these rotations in, and each triplet of rotated angles is called a set of **Euler angles**. In robotics, the triplet is **yaw**, **pitch** and **roll** (Z-Y-X).
+
+In the NUbots codebase, we denote a rotation matrix from space $b$ to space $a$ as `Rab`.
+
+The rotation matrices used for changing between spaces belong to the **special orthogonal group of order 3**. These all have the following properties
+
+$$
+\mathbf{R}^{-1} = \mathbf{R}^T
+$$
+
+$$
+\text{det}(\mathbf{R}) = 1
+$$
+
+The first property reduces the **complexity** of the computation to find the inverse matrix. The second property tells us this matrix can be inverted and it **preserves scale** between the two spaces.
+
+If we have a vector $\mathbf{r}_{CD}^a$, that is it goes from point D to point C in a space $a$, and we want to **rotate** it from space $a$ to space $b$, we write
+
+$$
+\mathbf{r}^b_{CD} = \mathbf{R}^b_a \cdot \mathbf{r}^a_{CD}
+$$
+
+This is written in the NUbots codebase as `rCDb = Rba * rCDa`. Note that the notation is $\mathbf{R}_{from}^{to}$.
+
+## Homogeneous Transformations
```

Comment:
We need to add somewhere that in Eigen we have to use `Eigen::Affine3[fd]` for these, as otherwise it'll yell at us when we do a 4x4 matrix with a 3d vector. Also using `Eigen::Affine3d` will mean that the matrix inverses are simpler since it doesn't have to do a full inverse.